[quant] QNNPACK deconvolution packing

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37497 [quant] QNNPACK factoring out common structs from conv/deconv
* #36790 [quant] QNNPACK deconv kernel and tests
* **#37405 [quant] QNNPACK deconvolution packing**



Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

Commit:
[quant] QNNPACK deconvolution packing

[ghstack-poisoned]

z-a-f commented:
This PR wil have errors until I combine the `conv_p` and `deconv_p` -- stay tuned

Commit:
Update on "[quant] QNNPACK deconvolution packing"




[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"


Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

z-a-f commented:
> This PR wil have errors until I combine the `conv_p` and `deconv_p` -- stay tuned

Resolved

Commit:
Update on "[quant] QNNPACK deconvolution packing"


Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"


Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"


Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

supriyar reviewed (approved):
LGTM. Please land the stack together once the entire flow works.

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]

Commit:
Update on "[quant] QNNPACK deconvolution packing"




Differential Revision: [D21301246](https://our.internmc.facebook.com/intern/diff/D21301246)

[ghstack-poisoned]
